### PR TITLE
fixed racecondition in stackall feature

### PR DESCRIPTION
**Pull request by Bellian**
Just added a stack queue.
Now waiting untill the RPC_StackResponse are done.

### CODE REVIEW

File: ValheimPlus/GameClasses/Inventory.cs [hunk 1]
```diff
@@ -130,47 +134,60 @@ private static void Prefix(Inventory __instance, ref bool message)
...
 
-            // try to stack all items on found containers
-            foreach (var container in chests)
-            {
-                if (container.m_inventory == __instance) continue;
-                container.StackAll();
+            dequeueContainer();
+        }
+
+        public static void dequeueContainer()
```
Comment (Grantapher) [inline ValheimPlus/GameClasses/Inventory.cs:148]: Lets move this logic to a dedicated static class. I think it can still be in this file though.

Comment (Grantapher) [inline ValheimPlus/GameClasses/Inventory.cs:148]: Idiomatically public static methods are upper camel case, `DequeueContainer`.

File: ValheimPlus/GameClasses/Inventory.cs [hunk 2]
```diff
@@ -197,4 +214,18 @@ public static IEnumerable<CodeInstruction> Transpiler(IEnumerable<CodeInstructio
         public static bool ContainsItemByName(Inventory inventory, string name) =>
             inventory.m_inventory.Any(item => !item.IsEquipable() && item.m_shared.m_name == name);
     }
+
+    // willkeep this here since the code is closely cuppled to the stack all inventory feature.
```
Comment (Grantapher) [inline ValheimPlus/GameClasses/Inventory.cs:218]: typos "Will keep", "coupled".

File: ValheimPlus/GameClasses/Inventory.cs [hunk 3]
```diff
@@ -121,7 +125,7 @@ public static class Container_StackAll_Patch
         [UsedImplicitly]
         private static void Prefix(Inventory __instance, ref bool message)
         {
-            if (!Configuration.Current.AutoStack.IsEnabled) return;
+            if (!true) return;
```
[the remark this answers is not in the document]
Comment (Bellian) [reply ValheimPlus/GameClasses/Inventory.cs:128]: damn. test code sorry! need to have a look again... sorry

File: ValheimPlus/GameClasses/Inventory.cs (hunk 2 above)
Comment (Bellian) [reply ValheimPlus/GameClasses/Inventory.cs:218]: My keyboard is dying...

**Comment by Bellian** [comment]
@Grantapher ping

**Pull request merged**